### PR TITLE
chore: add map component to platform

### PR DESCRIPTION
**Pull request by sissbruecker**
## Description

Adds the map component to the platform.

### CODE REVIEW

**Comment by ZheSun88** [comment]
I added the `vaadin-map` entry to **versions.json** too.. this is because we are still using the older way in platform for vaadin-component. 

have talked with Serhii about this, and we will probably remove it in later V23 minors. 


also, i guess the build will fail on the test coverage as this component has not been added to the test classes. let us see.

**Comment by ZheSun88** [comment]
I will fix the test for this PR. 

but i think we will not merge it before upgrading the project to V23.1-SNAPSHOT. As this feature is not part of 23.0, adding it to the vaadin-bom will make it totally public to the end users. 

if you need to use this compoment, you will still need to manually add the related dependency to your project.

**Comment by Artur-** [comment]
Why is this feature flagged component not part of 23? That would make it harder to get any feedback on it before removing the flag

**Comment by jouni** [comment]
Came here to say the same as Artur, that we want this in 23.0. This is different from how we’ve used to do things, but that’s changing, that experimental features / feature previews can be part of any platform release.

File: vaadin-platform-test/src/main/java/com/vaadin/platform/test/ComponentsView.java [hunk 1]
```diff
@@ -595,6 +595,8 @@ protected void setPresentationValue(String newPresentationValue) {
         MessageInput messageInput = new MessageInput();
         messageInput.addSubmitListener(e -> log.log(e.getValue()));
 
+        com.vaadin.flow.component.map.Map mapComponent = new com.vaadin.flow.component.map.Map();
```
Comment (manolo) [inline vaadin-platform-test/src/main/java/com/vaadin/platform/test/ComponentsView.java:598]: We should not have smoke tests for experimental components in the platform until they go to production

File: vaadin-platform-test/src/main/resources/vaadin-featureflags.properties [hunk 2]
```diff
@@ -0,0 +1,2 @@
+# Map component (Pro)
+com.vaadin.experimental.mapComponent=true
```
Comment (manolo) [inline vaadin-platform-test/src/main/resources/vaadin-featureflags.properties:2]: I'm doing some modifications in flow so as we don't need to enable experimental components in our test suite.
Blocking this for now

**Review by manolo — CHANGES_REQUESTED** [review]

File: vaadin-platform-test/src/main/resources/vaadin-featureflags.properties (hunk 2 above)
Comment (manolo) [reply vaadin-platform-test/src/main/resources/vaadin-featureflags.properties:2]: ok lets merge this, and fix flow later

**Review by manolo — APPROVED** [review]

**Pull request merged**